### PR TITLE
update storage interfaces

### PR DESCRIPTION
**Pull request by pavelkrolevets**
Changes according to https://github.com/bloxapp/ssv/pull/1020#discussion_r1271399377

### CODE REVIEW

File: operator/storage/storage.go [hunk 1]
```diff
@@ -27,7 +27,7 @@ var (
 type Storage interface {
 	// TODO: de-anonymize the sub-storages, like Shares() below
 
-	ROTxn() basedb.Txn
+	ROTxn() basedb.Reader
```
Comment (nkryuchkov) [inline operator/storage/storage.go:30]: Should `RWTxn` be `ReadWriter`?

File: storage/basedb/storage.go [hunk 2]
```diff
@@ -14,37 +14,41 @@ type Options struct {
...
+	Commit() error
+	Discard()
+}
+
+// Database interface for Badger DB
+type Database interface {
+	RWTxn() Txn
+	ROTxn() Reader // TODO: afaik there is no effect for Commit/Discard on read-only transactions so a `Reader` is sufficient?
+	ReadWriter
+	// TODO: consider moving these functions into Reader and ReadWriter interfaces?
```
Comment (nkryuchkov) [inline storage/basedb/storage.go:51]: Comment seems outdated

File: storage/kv/txn.go [hunk 3]
```diff
@@ -81,3 +81,19 @@ func (t badgerTxn) GetAll(prefix []byte, handler func(int, basedb.Obj) error) er
 func (t badgerTxn) Delete(prefix []byte, key []byte) error {
 	return t.txn.Delete(append(prefix, key...))
 }
+
+func (t badgerTxn) CountByCollection(prefix []byte) (int64, error) {
+	return t.db.CountByCollection(prefix)
```
Comment (nkryuchkov) [inline storage/kv/txn.go:86]: I think this should be a part of the transaction (same for methods below)

Comment (pavelkrolevets) [reply storage/kv/txn.go:86]: Do you mean on *badger.Txn? but it doesnt implement these methods. Can you please clarify a bit?

File: operator/storage/storage.go (hunk 1 above)
Comment (nkryuchkov) [reply operator/storage/storage.go:30]: I think Discard for both RO and RW, Commit only for RW @pavelkrolevets

File: storage/kv/txn.go (hunk 3 above)
Comment (nkryuchkov) [reply storage/kv/txn.go:86]: Yes, it doesn't, therefore @moshe-blox left them in the `basedb.IDb` interface (now `basedb.Database`). Can we remove them from `badgerTxn` and move these methods from transactions to `basedb.Database`?

Comment (pavelkrolevets) [reply storage/kv/txn.go]: Moved back to Database interface

**Comment by moshe-blox** [comment]
merged this draft and completed in #1020

**Pull request merged**